ktor-server/ktor-server-plugins/ktor-server-cors/jvmAndNix/src/io/ktor/server/plugins/cors/CORSConfig.kt

```diff
@@ -193,6 +198,13 @@ public class CORSConfig {
         allowHeader(HttpHeaders.XHttpMethodOverride)
     }
 
+    /**
+     * Allows using an origin matching [predicate] for the actual [CORS] request.
+     */
+    public fun allowOrigins(predicate: (String) -> Boolean) {
```

Comment:
Can you rename it to `allowHosts`?. It's confusing to use different terminology for one thing, even if `origin` fits better here.

Reply:
Actually I feel like in `https://example.org`, `origin` would be the entire thing while `host` would be just `example.org`, that's why I eventually renamed it to `allowOrigins` as I felt it was misleading to name it `allowHosts`

Reply:
Initially, I wanted to add both `allowHosts` and `allowOrigins` functions but did not continue this way as `allowHosts` would be a little more complicated (it would have `host` but also `scheme` as parameters at least, maybe more) and I personally don't have a use case for it.

Reply:
Ok, this makes sense.